newIDE/app/src/EventsSheet/index.js

```diff
@@ -1268,6 +1308,23 @@ export default class EventsSheet extends React.Component<Props, State> {
                             }}
                           />
                         )}
+                        {this.state.textEditorDialog && (
```

Comment:
Finally (if you read my previous comments), you can use `textEditedEvent` here:

```
{this.state.textEditedEvent && (
  <TextEditorDialog
    event={this.state.textEditedEvent}
    onApply={(changes) => {
      // Use the changes here and apply them to the event (this.state.textEditedEvent) - this will depend on the events type.
      // Then close the dialog:
      this.closeTextEditorDialog();
      // Surely call the method to update the history and refresh the events sheet.
    }}
    onClose={this.closeTextEditorDialog}
  />
)
```

Reply:
(updated my previous comment. TextEditorDialog should take care of updating the event BEFORE calling onApply).